src/Classes/ServiceAPI/MyRadio_Scheduler.php

```diff
@@ -96,22 +96,25 @@ public static function countPendingDisputes()
      *
      * @return int The new termid
      */
-    public static function addTerm($start, $descr)
+    public static function addTerm($start, $descr, $num_weeks)
     {
         if (date('D', $start) !== 'Mon') {
             throw new MyRadioException('Terms must start on a Monday.', 400);
         }
-
+        $num_weeks = (int)$num_weeks;
+        if (!is_int($num_weeks)){
+            throw new MyRadioException('Weeks must be an integer.', 400); 
+        }
         // Let's make this a GMT thing, exactly midnight
         $ts = gmdate('Y-m-d 00:00:00+00', $start);
-        $end = $start + (86400 * 68); // +68 days, to Friday of the 10th week
+        $end = $start + (86400 * ((7*$num_weeks)-2)); // to Friday of the 10th week
         $te = gmdate('Y-m-d 00:00:00+00', $end);
 
-        echo "INSERT INTO terms (start, finish, descr) VALUES ('$ts', '$te', '$descr') RETURNING termid";
+        echo "INSERT INTO terms (start, finish, descr, weeks) VALUES ('$ts', '$te', '$descr', '$num_weeks') RETURNING termid";
```

Comment:
I'm pretty sure this is why in testing I just got the raw SQL query printed to the screen when creating a new term instead of a rerendered scheduler with the _'Term ... has been added'_ success message. Maybe a debug line that was never removed all those years ago?

Probably best to just remove the line